media/src/main/java/com/yas/media/repository/FileSystemRepository.java

```diff
@@ -0,0 +1,72 @@
+package com.yas.media.repository;
+
+import com.yas.media.config.FilesystemConfig;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class FileSystemRepository {
+
+    private static final String DIRECTORY_DOES_NOT_EXIST = "Directory %s does not exist.";
+
+    private final FilesystemConfig filesystemConfig;
+
+    public String persistFile(String filename, byte[] content) throws IOException {
+
+        File directory = new File(filesystemConfig.getDirectory());
+        checkExistingDirectory(directory);
+        checkPermissions(directory);
+
+        Path filePath = buildFilePath(filename);
+        Files.write(filePath, content);
+        log.info("File saved: {}", filename);
+        return filePath.toString();
+    }
+
+    @SneakyThrows
+    public byte[] getFile(String filePath) {
```

Comment:
Consider returning an `InputStream` instead of a `byte[]` if we might be dealing with large files to avoid loading the entire file in memory.

Reply:
Which implementation should we use to return `InputStream`?

Reply:
I just committed to change byte[] into InputStream, thanks for your comment.